src/verify.c

```diff
@@ -882,15 +882,30 @@ static struct list *keep_matching_path(struct list *all_files)
 	struct list *matching_files = NULL;
 	struct list *iter = NULL;
 	struct file *file;
+	char *tmp_dir = NULL;
+
+	/* when considering the item as a directory, make sure the item
+	 * finishes with a forward slash so we can match items accuratelly,
+	 * oherwise we can end up with items that partially match */
+	tmp_dir = str_or_die("%s%s", cmdline_option_file, cmdline_option_file[str_len(cmdline_option_file)] == '/' ? "" : "/");
```

Comment:
I like the approach on treating all files as a possible directory here, but this is inconsistent with the messages  `Limiting diagnose to the following file/directory`. If there's a file type change you are printing 'directories' for files and files for directories.